runner/local/queueing_runner_test.go

```diff
@@ -0,0 +1,218 @@
+package local
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/runner/execer/execers"
+	"github.com/scootdev/scoot/runner/runners"
+	"github.com/scootdev/scoot/snapshot/snapshots"
+)
+
+const errorMsgFromRunner = "Error in fakeRunner Run()"
+
+// Send a run request that pauses, then another run request.
+// Verify that the status of the first run request is running, and
+// the status of the second request is queued.
+// Then send a signal to allow the first (paused) run request to finish.
+// Wait that all are run
+func TestQueueing2Messages(t *testing.T) {
+	env := setup(10, t)
+	defer env.teardown()
+	qr := env.qr
+
+	// send the first command - it should pause
+	run1 := assertRun(t, qr, running(), "pause", "complete 0")
+	run2 := assertRun(t, qr, pending(), "complete 1")
+	env.sim.Resume()
+	assertWait(t, qr, run1, complete(0), "n/a")
+	assertWait(t, qr, run2, complete(1), "n/a")
+}
+
+func TestQueueingMoreThanMaxMessage(t *testing.T) {
+	env := setup(4, t)
+	defer env.teardown()
+	qr := env.qr
+
+	var runIDs []runner.RunId
+
+	// block the runner
+	runID := assertRun(t, qr, running(), "pause", "complete 0")
+	runIDs = append(runIDs, runID)
+
+	// fill the queue
+	for i := 0; i < 4; i++ {
+		runID := assertRun(t, qr, pending(), "complete 0")
+		runIDs = append(runIDs, runID)
+	}
+
+	// overflow
+	_, err := qr.Run(&runner.Command{Argv: []string{"complete 5"}})
+	if err == nil || strings.Compare(QueueFullMsg, err.Error()) != 0 {
+		t.Fatal("Should not be able to schedule: ", err)
+	}
+
+	// resume
+	env.sim.Resume()
+
+	// drain
+	for _, id := range runIDs {
+		assertWait(t, qr, id, complete(0), "n/a")
+	}
+	runIDs = nil
+
+	// repeat
+	runID = assertRun(t, qr, running(), "pause", "complete 0")
+	runIDs = append(runIDs, runID)
+	for i := 0; i < 4; i++ {
+		runID := assertRun(t, qr, pending(), "complete 0")
+		runIDs = append(runIDs, runID)
+	}
+	_, err = qr.Run(&runner.Command{Argv: []string{"complete 5"}})
+	if err == nil || strings.Compare(QueueFullMsg, err.Error()) != 0 {
+		t.Fatal("Should not be able to schedule: ", err)
+	}
+	env.sim.Resume()
+	for _, id := range runIDs {
+		assertWait(t, qr, id, complete(0), "n/a")
+	}
+}
+
+func TestUnknownRunIdInStatusRequest(t *testing.T) {
+	env := setup(4, t)
+	defer env.teardown()
+	qr := env.qr
+
+	st, err := qr.Status(runner.RunId("not a real run id"))
+	if err == nil {
```

Comment:
do we want to verify that the error message was due to not finding the run id? (add: '|| !strings.Contains(err.Error(), runner.UnknownRunIdMsg')